tests/unit/plugins/janitor/test_reconciler.py

```diff
@@ -259,29 +255,34 @@ def extra_rrset():
 
     mock_get_records_for_zone_called = 0
 
+    return_rrsets = [rrset.copy() for rrset in fake_response_data['rrsets']]
+    return_rrsets[0]['rrdatas'] = ['10.4.5.6']
+    return_rrsets.append(extra_rrset)
+
     async def mock_get_records_for_zone(*args, **kwargs):
         nonlocal mock_get_records_for_zone_called
         mock_get_records_for_zone_called += 1
-        rrsets = fake_response_data['rrsets']
-        rrsets[0]['rrdatas'] = ['10.4.5.6']
-        rrsets.append(extra_rrset)
-        return rrsets
+        return return_rrsets
 
     recon_client.dns_client.get_records_for_zone = mock_get_records_for_zone
 
     input_rrsets = []
-    expected_missing_rrsets = []
-    expected_extra_rrsets = []
+    expected_missing_rrsets = set()
+    expected_extra_rrsets = set()
     if has_desired:
-        input_rrsets = rrsets
-        expected_missing_rrsets = [
+        for rrset in rrsets:
+            input_rrset = rrset.copy()
+            input_rrset['source'] = 'gceauthority'
+            input_rrsets.append(input_rrset)
+
+        expected_missing_rrsets = set(
             reconciler.ResourceRecordSet(**record)
             for record in [rrsets[0], soa_ns_rrsets[2]]
-        ]
-        expected_extra_rrsets = [
+        )
+        expected_extra_rrsets = set(
             reconciler.ResourceRecordSet(**record)
-            for record in [extra_rrset]
-        ]
+            for record in [return_rrsets[0], extra_rrset]
+        )
```

Comment:
This is the test that was updated to fail.  It seems we were reusing the same objects in many parts of the test and none of the objects had "source".  I first updated the test to reuse less, added "source" to the input and expected rrsets only (not to those returned from the GCE API) and ensure it failed.

After updating the API, I also had to adjust the expected returned extra rrsets here because of the decision to not drop deletions for updates, and replace lists with sets where necessary.